xExtension-WordHighlighter/extension.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+final class WordHighlighterExtension extends Minz_Extension
+{
+	const JSON_ENCODE_CONF = JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE | JSON_THROW_ON_ERROR;
+
+	public string $word_highlighter_conf = 'test';
+	public string $permission_problem = '';
+	public bool $enable_in_article = false;
+	public bool $enable_logs = false;
+	public bool $case_sensitive = false;
+	public bool $separate_word_search = false;
+
+	#[\Override]
+	public function init(): void
+	{
+		$this->registerTranslates();
+
+		// register CSS for WordHighlighter:
+		Minz_View::appendStyle($this->getFileUrl('style.css', 'css'));
+
+		Minz_View::appendScript($this->getFileUrl('mark.min.js', 'js'), false, false, false);
+
+		$current_user = Minz_Session::paramString('currentUser');
+
+		$staticPath = join_path($this->getPath(), 'static');
+		$configFileJs = join_path($staticPath, ('config.' . $current_user . '.js'));
+
+		if (file_exists($configFileJs)) {
+			Minz_View::appendScript($this->getFileUrl(('config.' . $current_user . '.js'), 'js'));
+		}
+
+		Minz_View::appendScript($this->getFileUrl('word-highlighter.js', 'js'));
+	}
+
+	#[\Override]
+	public function handleConfigureAction(): void
+	{
+		$this->registerTranslates();
+
+		$current_user = Minz_Session::paramString('currentUser');
+		$staticPath = join_path($this->getPath(), 'static');
+
+		$configFileJson = join_path($staticPath, ('config.' . $current_user . '.json'));
+
+		if (!file_exists($configFileJson) && !is_writable($staticPath)) {
+			$tmpPath = explode(EXTENSIONS_PATH . '/', $staticPath);
+			$this->permission_problem = $tmpPath[1] . '/';
+
+		} elseif (file_exists($configFileJson) && !is_writable($configFileJson)) {
+			$tmpPath = explode(EXTENSIONS_PATH . '/', $configFileJson);
+			$this->permission_problem = $tmpPath[1];
+
+		} elseif (Minz_Request::isPost()) {
+			$configWordList = html_entity_decode(Minz_Request::paramString('words_list'));
+
+			$this->word_highlighter_conf = $configWordList;
+			$this->enable_in_article = (bool) Minz_Request::paramString('enable-in-article');
+			$this->enable_logs = (bool) Minz_Request::paramString('enable_logs');
+			$this->case_sensitive = (bool) Minz_Request::paramString('case_sensitive');
+			$this->separate_word_search = (bool) Minz_Request::paramString('separate_word_search');
+
+			$lineSeparator = ((bool) strpos($configWordList, "\r\n")) ? "\r\n" : "\n";
```

Comment:
I was surprised that on my linux machine (under debian and under docker) I got `\r\n` as line separator - maybe related to PHP somehow?

Reply:
Got \r\n from where?

Reply:
from variable `$configWordList` which is initialized by:
`$configWordList = html_entity_decode(Minz_Request::paramString('words_list'));` (few lines above)

Reply:
And where did that come from? If it's a `<textarea>` for example those send \r\n.

Reply:
you are right, it's from a `<textarea>` so I can remove that line and use directly `"\r\n"` right?

Reply:
I'm not sure if all browsers are guaranteed to do that. Particularly I seem to recall Safari sending \n at some point. So I'd just keep it as is; doesn't hurt.

(Except I'd probably use something like `preg_split("/\r\n|\n|\r/", $etc)`. to keep it easier to read.)

Reply:
I replaced it in my [last commit](https://github.com/FreshRSS/Extensions/pull/238/commits/a928549d50bbb77b0d6e8dcc51a6f61f3585d6d0) with regex based on your suggestion. Thank you @Frenzie.